### PR TITLE
chore(url-state-provider): switch to vite/vitest

### PR DESCRIPTION
**Pull request by andypf**
This PR migrates the package from Jest to Vitest for unit testing, and transitions from rollup to Vite for faster builds and better developer experience. 

# Checklist

- [x] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [x] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings or errors.

### CODE REVIEW

**Review by barsukov — DISMISSED** [review]
LGTM please do not forget to close the related issue

**Comment by andypf** [comment]
closes #447

File: packages/url-state-provider/src/index.test.js [hunk 1]
```diff
@@ -110,199 +104,59 @@ describe("currentState", () => {
...
 
-      var urlState = new URL(window.location.href).searchParams.get("__s")
-      // The browsers allow 2040 characters long URLs.
-      // If we stay below 1500 characters, we can still support 50 different states with
-      // a length of up to 1000 characters per path.
-      expect(urlState.length < 1500).toEqual(true)
-    })
-  })
-
-  describe("replace", () => {
```
Comment (edda) [inline packages/url-state-provider/src/index.test.js:169]: Did you delete these tests on purpose? They seem relevant still.

**Review by edda — CHANGES_REQUESTED** [review]

File: packages/url-state-provider/src/index.test.js (hunk 1 above)
Comment (andypf) [reply packages/url-state-provider/src/index.test.js]: no, by accident :(
I've reverted the deletion and adopted the tests :)

**Review by barsukov — APPROVED** [review]

**Review by edda — APPROVED** [review]

**Review by hgw77 — APPROVED** [review]

**Pull request merged**